src/main/java/no/nav/tag/finnkandidatapi/tilgangskontroll/sts/STSClient.java

```diff
@@ -0,0 +1,62 @@
+package no.nav.tag.finnkandidatapi.tilgangskontroll.sts;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.*;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+@Slf4j
+@Component
+public class STSClient {
+
+    private final RestTemplate stsBasicAuthRestTemplate;
+    private final String stsUrl;
+
+    public STSClient(
+            RestTemplate stsBasicAuthRestTemplate,
+            @Value("${sts.url}") String stsUrl
+    ) {
+        this.stsBasicAuthRestTemplate = stsBasicAuthRestTemplate;
+        this.stsUrl = stsUrl;
+    }
+
+    public STStoken getToken() {
+        try {
+            ResponseEntity<STStoken> response = buildUriAndExecuteRequest();
+            if (response.getStatusCode() != HttpStatus.OK) {
+                String message = "Kall mot STS feiler med HTTP-" + response.getStatusCode();
+                log.error(message);
+                throw new RuntimeException(message);
+            }
+            return response.getBody();
```

Comment:
Dette er bare småpirk, men føler at denne returnen er litt gjemt og implisitt.
Hadde det vært mer leselig om man først sjekker for `response.getStatusCode() == HttpStatus.OK`, returnerer body inni der, og så kaster exception inni else?

Reply:
Enig i det, jeg ser litt på denne